### PR TITLE
redux migration - remove query actions/reducers

### PR DESCRIPTION
**Pull request by vutran**
First effort at migrating away from `redux` and `redux-saga`.

This only disconnects the `q` state from `redux`/`redux-saga` to keep it *small* and *manageable* for review.

Steps involved:

1. Disconnects `q` from `redux` and lift it up to the `AppContainer` level.
2. This is then passed down as a simple prop manually to the `QueryField`.
 - props ⬇️ : `AppContainer` -> `App` -> `QueryFieldContainer` -> `QueryField`.
 - events ⬆️  : `onQueryChange`, `onQueryReset` 

# When Context API is available.

Once `react` has been upgraded and the Context API is made available, we can simply remove the "prop drilling" of our `q` state and just send it down via a `Provider`.

For now, I think this is a great first step to slowly getting rid of `redux` and `redux-saga`.

### CODE REVIEW

File: app/renderer/src/sagas/index.js [hunk 1]
```diff
@@ -49,20 +46,6 @@ export function* handleSelectNextItem() {
   yield put(closeDetails());
 }
 
-/**
- * Handles the making a query
- */
-export function* handleQueryCommand(q) {
```
Comment (adnasa) [inline app/renderer/src/sagas/index.js:55]: yeeeeey, step one to get rid of this 🥇

**Review by adnasa — COMMENTED** [review]
looks like I caused the merge conflict for ya :bowtie: 
I'll sort that out as I review

File: app/renderer/src/components/App.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { compose, style } from 'glamor';
-import QueryFieldContainer from '../containers/QueryFieldContainer';
+import QueryFieldContainer from '../containers/query-field-container';
```
Comment (adnasa) [inline app/renderer/src/components/App.js:4]: 👍

File: app/renderer/src/components/App.js [hunk 3]
```diff
@@ -47,6 +53,8 @@ App.defaultProps = {
 
 App.propTypes = {
   theme: ThemeSchema,
```
Comment (adnasa) [inline app/renderer/src/components/App.js]: is `q` missing from the prop-type validation? it is being destructed in L28, after all.

File: app/renderer/src/components/App.js [hunk 4]
```diff
@@ -24,7 +25,7 @@ const base = style({
   overflow: 'hidden',
 });
 
-const App = ({ theme }) => {
+const App = ({ q, theme, onQueryChange, onQueryReset }) => {
```
Comment (adnasa) [inline app/renderer/src/components/App.js]: I would go as far as to remove the destruct
one thing I like with not using destruct, is that the forces SOC (cough propTypes validation)

File: app/renderer/src/containers/AppContainer.js [hunk 5]
```diff
@@ -29,19 +39,42 @@ const AppContainer = class extends Component {
     }
   }
 
+  updateQuery = q => {
+    this.setState({ q });
```
Comment (adnasa) [inline app/renderer/src/containers/AppContainer.js:43]: I'm sure you know this better than I do, but at which stage do we run a `debounce`?

File: app/renderer/src/containers/AppContainer.js [hunk 6]
```diff
@@ -29,19 +39,42 @@ const AppContainer = class extends Component {
...
+
+  resetQuery = () => {
+    // @todo - just clearn the results once results actions/reducers are removed
+    this.props.resetResults && this.props.resetResults();
+  };
+
   render() {
     const { theme } = this.props;
-    return <App theme={theme} />;
+    const { q } = this.state;
```
Comment (adnasa) [inline app/renderer/src/containers/AppContainer.js]: let's remove the destruct while we are here?

File: app/renderer/src/containers/AppContainer.js [hunk 7]
```diff
@@ -29,19 +39,42 @@ const AppContainer = class extends Component {
...
   }
 
+  updateQuery = q => {
+    this.setState({ q });
+    ipcRenderer.send(IPC_QUERY_COMMAND, { q });
+  };
+
+  resetQuery = () => {
+    // @todo - just clearn the results once results actions/reducers are removed
+    this.props.resetResults && this.props.resetResults();
```
Comment (adnasa) [inline app/renderer/src/containers/AppContainer.js:49]: `resetResults` is missing from propTypes?

File: app/renderer/src/containers/AppContainer.js [hunk 8]
```diff
@@ -29,19 +39,42 @@ const AppContainer = class extends Component {
...
+
+  // redux-actions
   setTheme: () => {},
 };
 
 AppContainer.propTypes = {
   theme: ThemeSchema,
+
+  // redux-actions
   setTheme: PropTypes.func,
```
Comment (adnasa) [inline app/renderer/src/containers/AppContainer.js:76]: hoh... at this stage, this is actually a `setState`, but let's deal with that later

**Review by adnasa — COMMENTED** [review]
- I took care of the merge conflict
- left some nitpicks regarding destruct
- question: where do we debounce?
- note: `setTheme` won't do anything after this PR, but I guess we can deal with that later

File: app/renderer/src/components/App.js (hunk 4 above)
Comment (vutran) [reply app/renderer/src/components/App.js]: Great point

File: app/renderer/src/containers/AppContainer.js (hunk 5 above)
Comment (vutran) [reply app/renderer/src/containers/AppContainer.js:43]: This is handled on the main process You're right though the client should debounce the query so we're not passing too many msgs to the main process.

**Comment by vutran** [comment]
> note: setTheme won't do anything after this PR, but I guess we can deal with that later

`setTheme` should still be working as `props.setTheme` has been binded from action creators at ~L85

**Comment by adnasa** [comment]
aah okay.

**Review by adnasa — COMMENTED** [review]
only this part needs to be addressed. the rest looks good to me 👍 
> resetResults is missing from propTypes?

File: app/renderer/src/containers/AppContainer.js (hunk 7 above)
Comment (vutran) [reply app/renderer/src/containers/AppContainer.js:49]: It's there :)

Comment (adnasa) [reply app/renderer/src/containers/AppContainer.js:49]: höh

**Review by adnasa — APPROVED** [review]

**Pull request merged**